python/gtsam/tests/test_Cal3Fisheye.py

```diff
@@ -0,0 +1,105 @@
+"""
+GTSAM Copyright 2010-2019, Georgia Tech Research Corporation,
+Atlanta, Georgia 30332-0415
+All Rights Reserved
+
+See LICENSE for the license information
+
+Cal3Unified unit tests.
+Author: Frank Dellaert & Duy Nguyen Ta (Python)
+"""
+import unittest
+
+import numpy as np
+
+import gtsam
+from gtsam.utils.test_case import GtsamTestCase
+
+
+class TestCal3Fisheye(GtsamTestCase):
+
+    def test_Cal3Fisheye(self):
+        K = gtsam.Cal3Fisheye()
+        self.assertEqual(K.fx(), 1.)
+        self.assertEqual(K.fy(), 1.)
+
+    def test_distortion(self):
+        "Equidistant fisheye model of focal length f, defined as r/f = tan(theta)"
+        equidistant = gtsam.Cal3Fisheye()
+        x, y, z = 1, 0, 1
+        u, v = equidistant.uncalibrate([x, y])
+        x2, y2 = equidistant.calibrate([u, v])
+        self.assertAlmostEqual(u, np.arctan2(x, z))
+        self.assertAlmostEqual(v, 0)
+        self.assertAlmostEqual(x2, x)
+        self.assertAlmostEqual(y2, 0)
+
+    def test_pinhole(self):
+        "Spatial equidistant camera projection"
+        x, y, z = 1.0, 0.0, 1.0
+        u, v = np.arctan2(x, z), 0.0
+        camera = gtsam.PinholeCameraCal3Fisheye()
```

Comment:
A lot of the review comments also apply here. Please ask for another round of reviews after updating these. :slightly_smiling_face: